src/components/ui/FfaListingComponent.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div class="columns-container">
+    <FfaListingHeader />
+    <FfaListingItem 
+      class="ffa-listing"
+      v-for="listing in displayedListings" 
+      :listing="listing" 
+      :key="listing.title" />
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import FfaListingItem from './FfaListingItem.vue'
+import FfaListingHeader from './FfaListingHeader.vue'
+import { MutationPayload } from 'vuex'
+import { getModule } from 'vuex-module-decorators'
+import FfaListingsModule from '../../vuexModules/FfaListingsModule'
+import FfaListing from '../../models/FfaListing'
+import '@/assets/style/components/listing.sass'
+import { FfaListingStatus } from '../../models/FfaListing'
+
+@Component({
+  components: {
+    FfaListingItem,
+    FfaListingHeader,
+  },
+})
+export default class FfaListingComponent extends Vue {
+  public ffaListingsModule: FfaListingsModule = getModule(FfaListingsModule, this.$store)
+  public displayedListings: FfaListing[] = []
+
+  @Prop()
+  public userAddress?: string
+
+  @Prop()
+  public status?: FfaListingStatus
+
+  private mounted() {
+    this.$store.subscribe(this.vuexSubscriptions)
+    this.handleDisplay()
+  }
+
+  private vuexSubscriptions(mutation: MutationPayload, state: any) {
+    switch (mutation.type) {
+      default:
+      this.handleDisplay()
+    }
+  }
+
+  private async handleDisplay() {
+    // Check if userAddress is truthy
+    const addressProvided = !!this.userAddress
+    const statusNotProvided = !!!this.status
+
+    // Show User listings only
+    if (addressProvided) {
```

Comment:
i know folks hate the switchception™ but i think these `if`s are messy and dangerous (a common refrain from the functional language camp).  i'd try rewriting the nested if's as nested switches (making use of `default:`) and seeing if that seems cleaner.  if not, feel free to put the `if`s back but i got $5 says it will generate at least one bug in the next 365 days.